daemon/containerd/cache.go

```diff
@@ -48,6 +47,24 @@ func (ic *imageCache) GetCache(parentID string, cfg *container.Config) (imageID
 		}
 	}
 
+	if parent == nil { // "FROM scratch"
+		imgs, err := ic.c.client.ImageService().List(ctx) // TODO is this right??
+		if err != nil {
+			return "", err
+		}
+		for _, img := range imgs {
+			id := img.Target.Digest.String()
+			image, err := ic.c.GetImage(ctx, id, imagetype.GetImageOpts{})
+			if err != nil {
+				return "", err
+			}
+			if isMatch(image, parent, cfg) {
```

Comment:
Hmmm, I just realized this isn't checking anything about `platform`, and we probably need to, right?